### ISSUE TITLE
Error with drawing lines in Cairo

### ISSUE THREAD

**Issue by larsiusprime** [openfl/openfl#909]
This regression is still happening with line drawing:

![limemiter3](https://cloud.githubusercontent.com/assets/739064/10821567/13f6ae0c-7e21-11e5-8fb1-f411f285354d.png)

There's two things wrong here:
1) Looks like the final joint is not rendering correctly on native targets (
This causes a bug in defender's quest where arrows that I draw for tutorials have a "bite" taken out of the arrowhead's tip)

2) The shapes are drawn in a different position than in Flash (though html5 and neko seem lined up to one another:)

![linemiter3_lines](https://cloud.githubusercontent.com/assets/739064/10821609/37f52a0e-7e21-11e5-9948-8fb3edeb3cb1.png)

The red line is for flash, the blue is the same lines from flash offset to line up with neko/html5. Notice that although the neko/html5 shapes are lined up with one another, that the spacing is off from flash as well, not just the overall offset position -- if the spacing was the same, the blue line would be on the left edge of the left triangle rather than cutting through it.

**Comment by jgranick** [comment]
I was sure that I fixed this before, but after testing both new and old versions (with the patch applied), I cannot see it fixed in Cairo. This is a shame. It does work in HTML5, perhaps someone could find a way to get the final joint to draw under Cairo

As for the coordinates, it is off because the miter joint is larger than 0.5 of the line width, in order to resolve this visually, I made the object bounds use the full width of the line in this case. Perhaps there is a better calculation we could use to make this work more exactly. As a result, getting the graphic bounds has a bit extra padding, and using those bounds (such as in this sample) will result in a bit more spacing. Without checking bounds, it should be drawn in the same position
